From 2d570b3f80b306ccdbc49b4a8e71c4945486fd1c Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Mon, 21 Aug 2023 23:53:47 +0200
Subject: [PATCH 1/1] qt,tests: Fix build in source directory, part 2

* lang/qt/tests/Makefile.am (AM_CPPFLAGS): Include Qt binding sources
before C++ binding sources and C sources.

This fixes the problem that the debug.h in the C sources was found
before the one in the Qt bindings.

Commit d23528cadf4284f508bc4ba05e1b5c14b46bd354 fixed it for Qt5, we
need Qt6 as well.
---
 lang/qt/tests/Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lang/qt/tests/Makefile.am b/lang/qt/tests/Makefile.am
index aedd3264..615daf6d 100644
--- a/lang/qt/tests/Makefile.am
+++ b/lang/qt/tests/Makefile.am
@@ -62,10 +62,10 @@ LDADD = ../../cpp/src/libgpgmepp.la ../src/libqgpgmeqt6.la \
         ../../../src/libgpgme.la @GPGME_QT6_LIBS@ @GPG_ERROR_LIBS@ \
         @GPGME_QT6TEST_LIBS@ @LDADD_FOR_TESTS_KLUDGE@ -lstdc++
 
-AM_CPPFLAGS = -I$(top_srcdir)/lang/cpp/src -I$(top_builddir)/src \
+AM_CPPFLAGS = -I$(top_srcdir)/lang/qt/src -I$(top_srcdir)/lang/cpp/src \
+              -I$(top_builddir)/src \
               @GPG_ERROR_CFLAGS@ @GPGME_QT6_CFLAGS@ @GPG_ERROR_CFLAGS@ \
               @LIBASSUAN_CFLAGS@ @GPGME_QT6TEST_CFLAGS@ -DBUILDING_QGPGME \
-              -I$(top_srcdir)/lang/qt/src \
               -DTOP_SRCDIR="$(top_srcdir)"
 endif
 
-- 
2.42.0

